smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/handlers/HttpHeaderMiddleware.kt

```diff
@@ -53,20 +54,24 @@ class HttpHeaderMiddleware(
                 .sortedBy { it.memberName }
             val prefixHeaderBindings = requestBindings
                 .filter { it.location == HttpBinding.Location.PREFIX_HEADERS }
+            val opIndex = OperationIndex.of(ctx.model)
+            val inputShape = opIndex.getInput(op).get()
+            val hasHeaders = MiddlewareShapeUtils.hasHttpHeaders(ctx.model, op)
+            if (headerBindings.isNotEmpty() || prefixHeaderBindings.isNotEmpty()) {
```

Comment:
Looks like the generation of the middleware is based on this line here -- we should have a single place for this logic so that we can use it here, as well as a place which decides whether or not to render it in the operation - i think you wrote this, but it does not seem to be used.

Reply:
corrected